install/uninstall.ps1

```diff
@@ -0,0 +1,125 @@
+}
+
+function Uninstall {
+    Close-AzureauthInstances
+
+    $locations = Get-AzureauthsInPath
+    Remove-AzureauthDirectories $locations
+    Remove-FromPath $locations
+
+    Write-Output "Uninstalled AzureAuth!"
+}
+
+function Get-AzureauthsInPath {
+    $allLocations = [System.Collections.ArrayList]@()
+    $azureauthsInPath = (Get-Command -Name azureauth -ErrorAction SilentlyContinue -CommandType Application -All).Source
+
+    # Uninstallation is only supported from the default AzureAuth location.
+    # We warn the user of any custom locations that are listed in the PATH.
+    # Installations in custom locations that are not listed in PATH cannot
+    # be found and uninstalled.
+    # We still...
+    ForEach($az in $azureauthsInPath) {
+        $tempDirectory = (Get-Item $az).Directory.FullName
+        if (!$az.Contains($azureauthDefaultLocation)) {
+            Write-Warning "Additional installation found in ${tempDirectory}"
+        } else {
+            # We add the PATH location to the list and not the default location to later
+            # remove it from the PATH (default location is the whole parent folder).
+            $_ = $allLocations.Add($tempDirectory)
+        }
+    }
+
+    return $allLocations
+}
+
+function Remove-AzureauthDirectories {
+    param([System.Collections.ArrayList]$locationsToDelete)
+
+    ForEach ($location in $locationsToDelete) {
+        Remove-InstallationFolder $location
+    }
+    
+    # We also delte AzureAuth parent folder
+    Remove-InstallationFolder $azureauthDefaultLocation
+}
+
+function Remove-InstallationFolder {
+    param ([string]$directory)
+
+    if (Test-Path -Path $directory) {
+        Write-Verbose "Removing installations at '${directory}'"
+        Remove-Item -Force -Recurse $directory
+    } else {
+        Write-Verbose "There were no installations found at '${directory}'"
+    }
+}
+
+function Remove-FromPath {
+    param ([System.Collections.ArrayList]$locationsToDelete)
+
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+
+    # Reconstruct the $PATH without any azureauth directories.
+    $updatedPath = "";
+    if (($null) -ne $currentPath) {
+        $paths = $currentPath.Split(";")
+        $pathArr = @()
+        ForEach($path in $paths){
+            if(!$path.Equals("") -And ($locationsToDelete.Count -eq 0 -Or !$locationsToDelete.Contains($path))) {
+                $pathArr += "${path}"
+            }
+            elseif (!$path.Equals("")) {
+                Write-Verbose "Removing '${path}' from `$env:PATH"
+            }
+        }
+        $updatedPath = ($pathArr -join ";") + ";"
+    }
+
+    Set-ItemProperty -Path $registryPath -Name PATH -Value $updatedPath
+    Send-SettingChange
+}
+
+function Close-AzureauthInstances {

```

Comment:
Umm, I am curious what others think. Should we rename this method to `Kill-AzureAuthInstances`? We use `taskkill`, so that might be more appropriate here.

Reply:
Kill does seem like a more appropriate verb. This is not a polite request. The `/f` flag is being used.

Reply:
Changed to kill